### PR TITLE
Replace app_identity.sign_blob with Cloud API call (#403).

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

File: src/appengine/libs/gcs.py [hunk 1]
```diff
@@ -35,6 +36,26 @@
...
+  service_account = 'projects/-/serviceAccounts/' + service_account_email()
+
+  response = iam.projects().serviceAccounts().signBlob(
+      name=service_account,
+      body={
+          'delegates': [],
+          'payload': base64.b64encode(data),
+      }).execute()
+
+  return base64.b64decode(response['signedBlob'])
```
Comment (inferno-chromium) [inline src/appengine/libs/gcs.py]: Do we want to verify signedBlob in response Or response is not None Or better error checking.

File: src/appengine/libs/gcs.py [hunk 2]
```diff
@@ -35,6 +36,26 @@
     ['url', 'bucket', 'key', 'google_access_id', 'policy', 'signature'])
 
 
+def service_account_email():
```
Comment (inferno-chromium) [inline src/appengine/libs/gcs.py]: This could be moved to utils as a helper.

File: src/appengine/libs/gcs.py (hunk 1 above)
Comment (inferno-chromium) [inline src/appengine/libs/gcs.py]: Isn't the response have signature field - https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts/signBlob ?

Comment (oliverchang) [reply src/appengine/libs/gcs.py]: That one is the deprecated API :) (See top of the page).

https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/signBlob is the correct documentation.

Comment (inferno-chromium) [reply src/appengine/libs/gcs.py]: My bad, ignore this then.

File: src/appengine/libs/gcs.py (hunk 2 above)
Comment (oliverchang) [reply src/appengine/libs/gcs.py]: Done.

File: src/appengine/libs/gcs.py (hunk 1 above)
Comment (oliverchang) [reply src/appengine/libs/gcs.py]: Wrapped this in a try

**Pull request merged**